com.jetbrains.GoLand.yaml

```diff
@@ -0,0 +1,57 @@
+app-id: com.jetbrains.GoLand
+runtime: org.freedesktop.Sdk
+runtime-version: '19.08'
+sdk: org.freedesktop.Sdk
+command: goland.sh
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --share=network
+  - --socket=pulseaudio
+  - --filesystem=host
+  - --socket=session-bus
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.freedesktop.secrets
+  - --filesystem=xdg-run/keyring
+  - --device=all
+  - --env=GOLAND_JDK=${FLATPAK_DEST}/goland/jbr/
+modules:
+  - shared-modules/glew/glew.json
+  - shared-modules/glu/glu-9.json
+
+  - name: freeglut
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DFREEGLUT_BUILD_STATIC_LIBS=OFF
+      - -DFREEGLUT_BUILD_DEMOS=OFF
+    sources:
+      - type: archive
+        url: https://prdownloads.sourceforge.net/freeglut/freeglut-3.2.1.tar.gz
+        sha256: d4000e02102acaf259998c870e25214739d1f16f67f99cb35e4f46841399da68
+
+  - name: goland
+    buildsystem: simple
+    build-commands:
+      - mkdir goland
+      - tar xzfv goland.tar.gz --directory=${FLATPAK_DEST}/ --strip-components=1
+      - install -T -Dm644 ${FLATPAK_DEST}/bin/goland.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
+      - install -t ${FLATPAK_DEST}/share/applications/ -Dm644 ${FLATPAK_ID}.desktop
+      - install -t ${FLATPAK_DEST}/share/appdata/ -Dm644 ${FLATPAK_ID}.appdata.xml
+      - icon_in="${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg"; icon_out="${FLATPAK_ID}.png";
+        for s in {32,64,128,256,512}; do rsvg-convert "${icon_in}" -w "${s}" -h "${s}"
+        -a -f png -o "${icon_out}"; install -p -Dm644 "${icon_out}" -t "${FLATPAK_DEST}/share/icons/hicolor/${s}x${s}/apps/";
+        done;
+    sources:
+      - type: file
+        url: https://download.jetbrains.com/go/goland-2020.2.1.tar.gz
+        sha256: 06cf8a7e6926d0aac2136ae1dde014f70ab71758e8b27b49519e8e47ac174d96
+        size: 424237985
+        x-checker-data:
+          type: jetbrains
+          code: PCP
+        dest-filename: goland.tar.gz
+      - type: file
+        path: com.jetbrains.GoLand.appdata.xml
```

Comment:
New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

Reply:
Ok, didn't fully understood but I renamed the `*.appdata.xml` to `*.metainfo.xml`and put it into `${FLATPAK_DEST}/share/metainfo/` folder